app/controllers/response_sets_controller.rb

```diff
@@ -64,8 +64,12 @@ def create
   # PATCH/PUT /response_sets/1/publish
   def publish
     if @response_set.status == 'draft'
-      @response_set.publish(current_user)
-      render :show, statis: :published, location: @response_set
+      if current_user.has_role?(:publisher)
+        @response_set.publish(current_user)
```

Comment:
In all the controllers, we should probably use @current_user everywhere instead of just 'current_user' for consistency.